seed/python-sdk/pagination/src/seed/users/client.py

```diff
@@ -101,7 +103,21 @@ def list_with_cursor_pagination(
             max_retries=request_options.get("max_retries") if request_options is not None else 0,  # type: ignore
         )
         if 200 <= _response.status_code < 300:
-            return pydantic_v1.parse_obj_as(ListUsersPaginationResponse, _response.json())  # type: ignore
+            _parsed_response = pydantic_v1.parse_obj_as(ListUsersPaginationResponse, _response.json())  # type: ignore
+            _has_next = False
+            _get_next = None
+            if _parsed_response.page is not None and _parsed_response.page.next is not None:
+                _parsed_next = _parsed_response.page.next.starting_after
+                _has_next = _parsed_next is not None
+                _get_next = lambda: self.list_with_cursor_pagination(
+                    page=page,
+                    per_page=per_page,
+                    order=order,
+                    starting_after=_parsed_next,
+                    request_options=request_options,
+                )
+            _items = _parsed_response.data
+            return SyncPaginator(has_next=_has_next, items=_items, get_next=_get_next)
```

Comment:
how does this work if `_parsed_next` and `_has_next` aren't lambdas and aren't recomputed on every page load?

Reply:
discussed offline, but the `get_next` is the API call wrapped in a lambda, so it's passing in the function call it was just in, with updated parameters
so the full SyncPager is recomputed and returned on the `get_next` call